include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -799,6 +799,12 @@ class __future : private std::tuple<_Args...>
     {
 #if !ONEDPL_ALLOW_DEFERRED_WAITING
         wait();
+#else
```

Comment:
Optional (but my preference):
If we are going to automatically check, then should we rename this `__checked_deferrable_wait`?

Reply:
Renamed